public/documentation/webcl.md

```diff
@@ -0,0 +1,284 @@
+JavaScript, which controls the execution of the kernel program by
+invoking the APIs defined by the WebCL specification.
+
+To implement the program above with WebCL, we will go through the
+following steps one by one:
+
+* Create a WebCL context.
+* Allocate buffers on your WebCL device. They will be used to hold the
+input and output data.
+* Create a WebCL program object. The program object contains the code to
+be executed on your WebCL device.
+* Create a WebCL kernel object. The kernel object designates the
+function to be called in the WebCL program, as well as the function
+arguments.
+* Kick off the computing by creating a command queue and enqueuing a
+series of commands into the command queue.
+
+# Creating a WebCL context
+
+A WebCL context is the collection of a lot of things. It designates the
+device(s) we are going to use, and it also manages other WebCL objects
+like memory buffers, command queues and WebCL programs. The following
+shows how to create a WebCL context which utilizes GPU as the backend:
+
+    var context = webcl.createContext(webcl.DEVICE_TYPE_GPU);
+
+In case the GPU is not OpenCL-capable, we also have a try on CPU, and if
+that fails again, an exception will be thrown. Here is a more robust
+context creation:
+
+    var context;
+    try {
+      context = webcl.createContext(webcl.DEVICE_TYPE_GPU);
+    } catch (e) {
+      try {
+        context = webcl.createContext(webcl.DEVICE_TYPE_CPU);
+      } catch (e) {
+        alert("WebCL is not supported by your device!");
+        throw e;
+      }
+    }
+
+# Allocating input and output buffers
+
+For the sake of simplicity, we still use the sequential way to construct
+our input array:
+
+    var len = 1000000;
+    var i;
+    var arr1 = new Float32Array(len);
+    for (i = 0; i < len; i++)
+      arr1[i] = i;
+
+However, the variable `arr1` cannot be accessed directly by the kernel
+program we are going to create soon. A buffer needs to be allocated in
+this WebCL context and initialized with data in `arr1`:
+
+    var buf1 = context.createBuffer(webcl.MEM_READ_ONLY, arr1.byteLength, arr1);
+
+The constant `webcl.MEM_READ_ONLY` means that the created buffer will be
+read-only for the kernel program, so that specific optimizations can be
+performed based on this hint.
+
+Similarly, kernel program cannot write to a JavaScript typed array
+directly. We must create another buffer as an intermediate holder for
+the result:
+
+    var buf2 = context.createBuffer(webcl.MEM_WRITE_ONLY, arr1.byteLength);
+
+You must have guessed what the first argument `webcl.MEM_WRITE_ONLY`
+means. Also, we do not provide a third argument here since there's no
+need to initialize the output buffer with an existing typed array.
+
+# Creating the WebCL kernel program object
+
+To parallelize the calculation, we divide the task into 1,000,000 **work
+item**s. Each work item is assigned with a unique ID starting from 0,
+and it just calculates the square root of one element in the input
```

Comment:
I think you need a space after the closing ** or the bold section won't end here